src/plots/cartesian/autorange.js

```diff
@@ -226,8 +226,13 @@ function makePadFn(fullLayout, ax, max) {
     var A = 0;
     var B = 0;
     if(!isLinked(fullLayout, ax._id)) {
-        A = padInsideLabelsOnAnchorAxis(ax, max);
-        B = padInsideLabelsOnThisAxis(ax, max);
+        if(ax.rangemode !== 'nonnegative') {
```

Comment:
I don't think this is really the condition we want - we're not trying to completely prevent extra padding for labels when `rangemode` is `'nonnegative'`, we just want to prevent the range from going past zero. ie we never want this to prevent the high end of the range from expanding to make room for the labels, nor do we want to prevent the low end from expanding _unless it's going to go past zero_ - so for example if you have for example scatter points from 1 to 10 and there are inside labels on the low end, so that without labels the range would be something like `[0.4, 10.6]`, the labels should expand the range down to zero but not beyond.